core/node/builder.cpp

```diff
@@ -0,0 +1,330 @@
+    }
+
+    OUTCOME_TRY(initNetworkName(o.chain_db->genesisTipset(), o.ipld, config));
+    log()->info("Network name: {}", config.network_name);
+    log()->info("Genesis: {}", config.genesis_cid.value().toString().value());
+
+    log()->debug("Creating host...");
+
+    // config.gossip_config.protocol_version = "/floodsub/1.0.0";
+    // config.gossip_config.D_max = config.gossip_config.D_min = 0;
+
+    auto injector = libp2p::injector::makeHostInjector<
+        boost::di::extension::shared_config>(
+        //        boost::di::bind<libp2p::security::SecurityAdaptor
+        //        *[]>().template
+        //        to<libp2p::security::TlsAdaptor>()[boost::di::override],  //
+        //        NOLINT
+
+        boost::di::bind<clock::UTCClock>.template to<clock::UTCClockImpl>());
+
+    o.io_context = injector.create<std::shared_ptr<boost::asio::io_context>>();
+
+    o.scheduler =
+        injector.create<std::shared_ptr<libp2p::protocol::Scheduler>>();
+
+    o.host = injector.create<std::shared_ptr<libp2p::Host>>();
+
+    o.utc_clock = injector.create<std::shared_ptr<clock::UTCClock>>();
+
+    auto identify_protocol =
+        injector.create<std::shared_ptr<libp2p::protocol::Identify>>();
+    auto identify_push_protocol =
+        injector.create<std::shared_ptr<libp2p::protocol::IdentifyPush>>();
+    auto identify_delta_protocol =
+        injector.create<std::shared_ptr<libp2p::protocol::IdentifyDelta>>();
+
+    log()->debug("Creating peer manager...");
+
+    o.peer_manager =
+        std::make_shared<fc::sync::PeerManager>(o.host,
+                                                o.utc_clock,
+                                                identify_protocol,
+                                                identify_push_protocol,
+                                                identify_delta_protocol);
+
+    log()->debug("Creating chain loaders...");
+
+    o.blocksync_client =
+        std::make_shared<fc::sync::blocksync::BlocksyncClient>(o.host, o.ipld);
+
+    o.block_loader = std::make_shared<fc::sync::BlockLoader>(
+        o.ipld, o.scheduler, o.blocksync_client);
+
+    o.tipset_loader =
+        std::make_shared<fc::sync::TipsetLoader>(o.scheduler, o.block_loader);
+
+    //    auto weight_calculator =
+    //        std::make_shared<blockchain::weight::WeightCalculatorImpl>(o.ipld);
+    //
+    //    auto power_table = std::make_shared<power::PowerTableImpl>();
+    //
+    //    auto bls_provider = std::make_shared<crypto::bls::BlsProviderImpl>();
+    //
+    //    auto secp_provider =
+    //        std::make_shared<crypto::secp256k1::Secp256k1ProviderImpl>();
+    //
+    //    // TODO: persistent keystore
+    //    auto key_store =
+    //    std::make_shared<storage::keystore::InMemoryKeyStore>(
+    //        bls_provider, secp_provider);
+    //
+
+    o.vm_interpreter = std::make_shared<vm::interpreter::CachedInterpreter>(
+        std::make_shared<vm::interpreter::InterpreterImpl>(), o.kv_store);
```

Comment:
Need to pass map with namespaced/prefixed keys, to prevent collisions when components use same keys